lib/services/extensibility-service.ts

```diff
@@ -0,0 +1,135 @@
+import * as path from "path";
+import { cache } from "../common/decorators";
+import * as constants from "../constants";
+
+export class ExtensibilityService implements IExtensibilityService {
+	private get pathToExtensions(): string {
+		return path.join(this.$options.profileDir, "extensions");
```

Comment:
I think the time is ripe to move `this.$options.profileDir` into a separate method/service or whatnot and get it from here because we'd really like to keep our services clear of `$options`.

As this requires some additional effort in finding all of the occurrences of `this.$options.profileDir` where the new method may be used I'd like to see this in a separate PR I think, ping @rosen-vladimirov for an opinion